src/stories/Drawer.stories.js

```diff
@@ -0,0 +1,96 @@
+import BasicDrawer from '../components/drawer/BasicDrawer.vue'
+
+// This is required for each story
+export default { title: 'Drawer' }
+
+// Customize components here.
+export const basicDrawer = () => ({
+  components: { BasicDrawer },
+  data () {
+    return {
+      drawer: false,
+      direction: 'rtl',
+      title: 'I am the title'
+    }
+  },
+  watch: {
+    drawer (newValue, oldValue) {
+      if (!newValue) {
+        this.drawer = false
+      }
+    }
+  },
+  template: `
+  <div>
+    <el-button @click="drawer = true" type="primary" style="margin-left: 16px;">
+      Open drawer
+    </el-button>
+    <basic-drawer
+      :title="title"
+      :drawerVisible="$data._drawer"
```

Comment:
@michaelbukachi is this how we add inner variable? sorry for all the back and forth :( I am very new to all this.

Reply:
Nope. It should be:
```
:drawerVisible="_drawer"
```
then somewhere in the script section under the data field:
```
data: function () {
  return {
    _drawer: false
  }
}
```